app/controllers/tasks/collection_objects/area_and_date_controller.rb

```diff
@@ -0,0 +1,137 @@
+class Tasks::CollectionObjects::AreaAndDateController < ApplicationController
+  include TaskControllerConfiguration
+
+  # GET
+  def index
+    @geographic_areas         = GeographicArea.where('false')
+    @collection_objects       = CollectionObject.where('false')
+    @collection_objects_count = 0
+    @early_date               = CollectionObject.where(project: sessions_current_project_id).order(:created_at).limit(1).pluck(:created_at).first
+    if @early_date.blank?
+      @early_date = Date.parse('1700/01/01')
+    end
+    @late_date = CollectionObject.where(project: sessions_current_project_id).order(created_at: :desc).limit(1).pluck(:created_at).first
+    if @late_date.blank?
+      @late_date = Date.today
+    end
+  end
+
+  # POST
+  # find all of the objects within the supplied area and within the supplied data range
+  def find
+    message             = ''
+    # find the objects in the selected area
+    @geographic_area_id = params[:geographic_area_id]
+    @geographic_area    = GeographicArea.find(@geographic_area_id) unless @geographic_area_id.blank?
+    @shape_in           = params[:drawn_area_shape]
+    set_and_order_dates(params)
+
+    if @shape_in.blank? and @geographic_area_id.blank? # missing "? " was fixed
```

Comment:
Move logic to a model, likely CollectionObject?!  E.g. ```area_object_ids = CollectionObject.geographic_items_by_area_or(?)_shape(geographic_area_id, shape)``` Why does the ```if``` return a scope or an Array of IDs? Return same type?